packages/fx-core/src/question/create.ts

```diff
@@ -497,8 +527,53 @@ export class CapabilityOptions {
         ];
   }
 
-  static officeAll(): OptionItem[] {
-    return [...CapabilityOptions.officeAddinItems(), CapabilityOptions.officeAddinImport()];
+  static officeAddinStaticCapabilities(host?: string): OptionItem[] {
+    const items: OptionItem[] = [];
+    for (const h of Object.keys(OfficeAddinProjectConfig)) {
+      if (host && h !== host) continue;
+      const hostValue = OfficeAddinProjectConfig[h];
+      for (const capability of Object.keys(hostValue)) {
+        const capabilityValue = hostValue[capability];
+        items.push({
+          id: capability,
+          label: getLocalizedString(capabilityValue.title),
+          detail: getLocalizedString(capabilityValue.detail),
+        });
+      }
+    }
+    return items;
+  }
+
+  static officeAddinDynamicCapabilities(projectType: string, host?: string): OptionItem[] {
+    const items: OptionItem[] = [];
+    if (
+      projectType === ProjectTypeOptions.outlookAddin().id ||
+      projectType === ProjectTypeOptions.officeAddin().id ||
+      host === OfficeAddinHostOptions.outlook().id
+    ) {
+      const capabilityValue = OfficeAddinProjectConfig.json["json-taskpane"];
+      items.push({
+        id: "json-taskpane",
+        label: getLocalizedString(capabilityValue.title),
+        detail: getLocalizedString(capabilityValue.detail),
+      });
+      if (projectType === ProjectTypeOptions.outlookAddin().id) {
+        items.push(CapabilityOptions.outlookAddinImport());
+      }
```

Comment:
there may be an item missing:
`if (projectType === ProjectTypeOptions.officeAddin().id) {
        items.push(CapabilityOptions.officeAddinImport());
      }`